lib/command.js

```diff
@@ -1257,81 +1449,93 @@ Expecting one of '${allowedValues.join("', '")}'`);
    */
 
   _parseCommand(operands, unknown) {
+    let chain;
+
     const parsed = this.parseOptions(unknown);
     this._parseOptionsEnv(); // after cli, so parseArg not called on both cli and env
     this._parseOptionsImplied();
+    chain = this._chainOrCall(chain, () => this._awaitOptionResavePromises());
```

Comment:
Just a comment. 

This is a nice logical position for processing the options just after they are defined. It does unfortunately spread the chaining logic through the top of this routine. I went for positions close to the hooks since I knew there was already chaining support for hooks.

(And processing async parseArgs for arguments when is no action handler spreads the chaining logic down through routine.)

Reply:
> It does unfortunately spread the chaining logic through the top of this routine.

But on the other hand, it unifies the option value awaiting for parent and leaf commands, and `_dispatchSubcommand()` can remain unchanged. Also, it is actually quite nice to have the `chain` declaration at the very top and `return chain` at the bottom, that kind of sets the logical frame for the function.